src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -67,75 +63,64 @@ public BtToRequestSocket(BluetoothSocket socket, BtSocketDisconnectedCallBack ha
         mmOutStream = tmpOut;
     }
 
-    public void setPort(int port) {
-        mHTTPPort = port;
-    }
-
     public void run() {
+
         try {
             print_debug("start Socket with port: " + mHTTPPort);
+            mHTTPPort = 0; // just to make sure its zero
```

Comment:
You already set this in the constructor. Why set it again here? You already know it's zero.

Also, how do you protect yourself against Run being called twice? I would just have a 'running' boolean (note the primitive type, not the class Boolean) marked as volatile.